src/solve.jl

```diff
@@ -56,25 +56,26 @@ end
 solve(system::AbstractSystem, options::Pair{Symbol,<:Any}...) =
     solve(system, Options(Dict{Symbol,Any}(options)))
 
-function solve!(system::InitialValueProblem{<:Union{AbstractContinuousSystem,
-                                                     AbstractDiscreteSystem}},
+function solve!(problem::InitialValueProblem,
```

Comment:
Why remove this?

Reply:
I thought about keeping the `solve!` interface generic (it's already the case for the other `solve` methods).